aiohttp/cookiejar.py

```diff
@@ -52,9 +55,24 @@ class CookieJar(AbstractCookieJar):
 
     DATE_YEAR_RE = re.compile(r"(\d{2,4})")
 
-    MAX_TIME = datetime.datetime.max.replace(tzinfo=datetime.timezone.utc)
-
-    MAX_32BIT_TIME = datetime.datetime.fromtimestamp(2**31 - 1, datetime.timezone.utc)
+    # calendar.timegm() fails for timestamps after datetime.datetime.max
+    # Minus one as a loss of precision occurs when timestamp() is called.
+    MAX_TIME = (
+        int(datetime.datetime.max.replace(tzinfo=datetime.timezone.utc).timestamp()) - 1
+    )
+    try:
+        calendar.timegm(time.gmtime(MAX_TIME))
+    except OSError:
+        # Hit the maximum representable time on Windows
+        # https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/localtime-localtime32-localtime64
+        MAX_TIME = calendar.timegm((3000, 12, 31, 23, 59, 59, -1, -1, -1))
+        # The below line should never raise an exception
+        time.gmtime(MAX_TIME)
```

Comment:
Why is this line here? Should we just remove it?

Reply:
We assume that OSError will only occur on Windows, without really verifying the current OS. The line is to ensure if there is any other (OS, arch) that raises OSError, the Windows-specific max time won't be too large.

Reply:
But, if it does happen, then the whole library fails on import, even if the user would never have received a value anywhere near the MAX_TIME and wouldn't have experienced any issues at runtime.

Reply:
OK then. I've removed the line